src/plugins/data/public/query/query_string/language_service/get_query_control_links.tsx

```diff
@@ -0,0 +1,172 @@
+    ariaLabel: i18n.translate('discover.queryControls.languageReference', {
+      defaultMessage: `Language Reference`,
+    }),
+    run: async (anchorElement) => {
+      if (props.queryLanguage === 'PPL' || props.queryLanguage === 'SQL') {
+        const flyoutSession = props.services.overlays!.openFlyout(
+          toMountPoint(
+            <OpenSearchDashboardsContextProvider services={props.services}>
+              <PPLReferenceFlyout
+                onClose={() => flyoutSession?.close?.().then()}
+                makeUrl={(searchId: any) => `#/view/${encodeURIComponent(searchId)}`}
+              />
+            </OpenSearchDashboardsContextProvider>
+          )
+        );
+      } else {
+        if (isLanguageReferenceOpen) {
+          onCloseLanguageReference();
+          return;
+        }
+
+        setIsLanguageReferenceOpen(true);
+        document.body.appendChild(languageReferenceContainer);
+
+        const element = (
+          <EuiWrappingPopover
+            id="languageReferencePopover"
+            button={anchorElement}
+            isOpen={true}
+            closePopover={onCloseLanguageReference}
+            panelPaddingSize="s"
+            anchorPosition="downLeft"
+            anchorClassName="euiFormControlLayout__append"
+          >
+            <EuiPopoverTitle>
+              <FormattedMessage
+                id="data.query.queryBar.syntaxOptionsTitle"
+                defaultMessage="Syntax options"
+              />
+            </EuiPopoverTitle>
+            <div style={{ width: '350px' }}>
+              <EuiText size="s">
+                <p>
+                  <FormattedMessage
+                    id="data.query.queryBar.syntaxOptionsDescription"
+                    defaultMessage="The {docsLink} (DQL) offers a simplified query
+              syntax and support for scripted fields."
+                    values={{
+                      docsLink: (
+                        <EuiLink href={osdDQLDocs} target="_blank">
+                          {dqlFullName}
+                        </EuiLink>
+                      ),
+                    }}
+                  />
+                </p>
+              </EuiText>
+            </div>
+          </EuiWrappingPopover>
+        );
+
+        ReactDOM.render(element, languageReferenceContainer);
+      }
+    },
+    iconType: 'iInCircle',
+  };
+
+  const languageToggle: QueryControl = {
+    id: 'languageToggle',
+    label: i18n.translate('discover.queryControls.languageToggle', {
+      defaultMessage: 'Toggle',
+    }),
+    testId: 'languageToggle',
+    ariaLabel: i18n.translate('discover.queryControls.languageToggle', {
+      defaultMessage: `Language Toggle`,
+    }),
+    run: () => {
+      setIsCollapsed(!isCollapsed);
+      props.onToggleCollapse();
+    },
+    iconType: isCollapsed ? 'expand' : 'minimize',
+  };
+
+  const queryControls =
+    props.queryLanguage === 'PPL' || props.queryLanguage === 'SQL'
```

Comment:
Lets not check explicitly for the PPL/SQL stuff here. The expand should be associated with the editor itself. If its the DQL editor, it should have this but if its the Default editor, then it should have this control alongside the other controls